File: mhchain.cpp
/*
AMChain is trying for creating peer-to-peer
decentraliztion virtual super-computing soft and hard <\\>ware
a Marxist application perfomance interface
with neurologic implemetation
for Blockchain Bigdata analysing

2022 (c) AMChain

MHCHain is Meta Humanist blockchain realization for AMChain Protocol

This code is RaW-PrImEr for our humanity future generations

перед вами реализация на языке C++
классовая модель основного Атома (блока)
работающего в AMChain
*/


//    код работает на gcc в lubuntu 22.04
// 
#include <string>


struct Blockchain
{
   std::string index; // index блока
   std::string timestamp; // ага богаты на ОЗУ храним всё в стринг
   std::string data; //data for current block
   std::string prev_hash; // stores hah for previous block
   std::string other; // доп данные еее

   std::string Hi() {
         std::string r ="HHH";
      return r;
   }
};

int main() {
   Blockchain blockchain;
   printf(blockchain);
}

File: mhchain.php
<?php

/*
AMChain is trying for creating peer-to-peer
decentraliztion virtual super-computing soft and hard <\\>ware
a Marxist application perfomance interface
with neurologic implemetation
for Blockchain Bigdata analysing

2022 (c) AMChain

MHCHain is Meta Humanist blockchain realization for AMChain Protocol

This code is RaW-PrImEr for our humanity future generations

перед вами реализация на языке PHP
классовая модель основного Атома (блока)
работающего в AMChain
*/


// 	код работает на
//	PHP 8.1.2 (cli) (built: Jun 13 2022 13:52:54) (NTS)


class Blockchain {
	public  $index, $timestamp, $data, $previousHash, $other;

	function Hi() {
		echo "Приветик, Мой Хуёвый Блокчейн";
	}

	function md5Hash() {
		$r = md5($this->index . $this->timestamp . $this->data . $this->previousHash . $this->other);
		return $r;
	}

}

$blockchain = new Blockchain();
$blockchain->index = 0;
$blockchain->timestamp=time();
$blockchain->data="BLOB8BLOB";
$blockchain->previousHash=md5("122887");
$blockchain->Hi();

echo "\n BEGIN \n APP started here \n";

echo "HASH previous block is: " . $blockchain->md5Hash() . "\n";
print_r($blockchain);

?>
